internal/armada/repository/usage.go

```diff
@@ -161,6 +167,41 @@ func (r *RedisUsageRepository) UpdateClusterLeased(report *api.ClusterLeasedRepo
 	return err
 }
 
+func (r *RedisUsageRepository) GetClusterQueueResourceUsage() (map[string]*schedulerobjects.ClusterResourceUsageReport, error) {
+	result, err := r.db.HGetAll(clusterQueueResourceUsageKey).Result()
+	if err != nil {
+		return nil, fmt.Errorf("[RedisUsageRepository.GetClusterQueueResourceUsage] error reading from database: %s", err)
+	}
+	reports := make(map[string]*schedulerobjects.ClusterResourceUsageReport)
+
+	for k, v := range result {
+		report := &schedulerobjects.ClusterResourceUsageReport{}
+		err = proto.Unmarshal([]byte(v), report)
+		if err != nil {
+			return nil, fmt.Errorf("[RedisUsageRepository.GetClusterQueueResourceUsage] error unmarshalling: %s", err)
+		}
+		reports[k] = report
+	}
+	return reports, nil
+}
+
+func (r *RedisUsageRepository) UpdateClusterQueueResourceUsage(cluster string, resourceUsage *schedulerobjects.ClusterResourceUsageReport) error {
+
+	data, err := proto.Marshal(resourceUsage)
+	if err != nil {
+		return fmt.Errorf("[RedisUsageRepository.UpdateClusterQueueResourceUsage] error marshalling report: %s", err)
+	}
+
+	pipe := r.db.TxPipeline()
+	pipe.HSet(clusterQueueResourceUsageKey, cluster, data)
```

Comment:
Can we expire these automatically? In case a cluster goes away we'd like the reports to be expired.

Reply:
Also, doesn't this need to somehow be keyed by both executor and queue? Since there are many per-queue reports per executor.

Reply:
`clusterQueueResourceUsageKey` is a map like object where `cluster` is the key and `data` the value.  In this case the `data` is itlsef a map  of queue -> cluster so I think we're good.

I don't think we want this key to expire, rather `Data` contains a timestamp saying when it was last updated.  If the time is too far in the past we don't consider the usage valid and therefore exclude it from scheduling (this is consistent with how we do things in the legacy usage)